Simplified/ErrorHandling/NYPLAlertUtils.swift

```diff
@@ -197,3 +197,42 @@ import UIKit
     }
   }
 }
+
+extension NYPLAlertUtils {
+  /**
+   Generates an alert view that presents another alert when being dismissed
+   - Parameter announcements: an array of announcements that goes into alert message.
+   - Returns: The alert controller to be presented.
+   */
+  class func alert(announcements: [Announcement]) -> UIAlertController? {
+    let title = NSLocalizedString("Announcement", comment: "")
+    var currentAlert: UIAlertController? = nil
+    
+    let alerts = announcements.map {
+      UIAlertController.init(title: title, message: $0.content, preferredStyle: .alert)
+    }
+    
+    // Present another alert when the current alert is being dismiss
+    // Add the presented announcement to the presentedAnnouncement document
+    for (i, alert) in alerts.enumerated() {
+      if i > 0 {
```

Comment:
This `if` statement is only called when there is more than 1 announcement, and we present the next announcement when the current one being dismissed.
Otherwise, we just `addPresentedAnnouncement(id:)` without presenting another announcement which happens in the `if let` statement below.
I can use `currentAlert != nil` instead of `i < 0` if it makes the logic more understandable.